modules/core/core-app/src/main/java/com/enonic/xp/core/impl/app/ApplicationFactory.java

```diff
@@ -46,27 +50,33 @@ public ApplicationImpl create( final Bundle bundle, final Configuration config )
 
     ApplicationUrlResolver createUrlResolver( final Bundle bundle )
     {
-        final ApplicationUrlResolver bundleUrlResolver = new BundleApplicationUrlResolver( bundle );
-        final ApplicationUrlResolver nodeResourceApplicationResolver =
+        final BundleApplicationUrlResolver bundleUrlResolver = new BundleApplicationUrlResolver( bundle );
+        final NodeResourceApplicationUrlResolver nodeResourceApplicationResolver =
             new NodeResourceApplicationUrlResolver( ApplicationKey.from( bundle ), nodeService );
+        final ClassLoaderApplicationUrlResolver classLoaderUrlResolver = createClassLoaderUrlResolver( bundle );
+
+        final boolean addCLR = RunMode.DEV.equals( this.runMode ) && classLoaderUrlResolver != null;
 
-        if ( this.runMode != RunMode.DEV )
+        if ( appConfig.virtual_enabled() )
         {
-            return new MultiApplicationUrlResolver( nodeResourceApplicationResolver, bundleUrlResolver );
+            if ( appConfig.virtual_schema_override() )
+            {
+                return addCLR
+                    ? new MultiApplicationUrlResolver( nodeResourceApplicationResolver, classLoaderUrlResolver, bundleUrlResolver )
+                    : new MultiApplicationUrlResolver( nodeResourceApplicationResolver, bundleUrlResolver );
+            }
+            else
+            {
+                return new RealOverVirtualApplicationUrlResolver( bundleUrlResolver, classLoaderUrlResolver, nodeResourceApplicationResolver );
```

Comment:
use `MultiApplicationUrlResolver` for bundleUrlResolver and  classLoaderUrlResolver